pallet-chainlink-feed/src/lib.rs

```diff
@@ -251,14 +250,23 @@ pub mod pallet {
 		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
 		/// Type for feed indexing.
-		type FeedId: Member + Parameter + Default + Copy + HasCompact + BaseArithmetic;
+		type FeedId: Member
+			+ Parameter
+			+ Default
+			+ Copy
+			+ HasCompact
+			+ BaseArithmetic
+			+ AccountIdConversion<Self::AccountId>;
```

Comment:
This simply puts the responsibility of setting up feed account to the user

I'm still not sure about the safety of this approach, are there any best practices in substrate for this kind of thing?

However if we only use a single account for payments (derived from palletId) and use the `payment` field within a `Feed` merely as tracker, this should become a non issue.

Reply:
This line still exists is that I forgot to revert all commits in the last commit

Please don't rush as I just start to refactor this PR right now, you just repeated what you said last week LOL

Reply:
The safety of converting feeds into accounts is if the feed id will be duplicate.

The `FeedId` are generated by `FeedCounter` which means if the `FeedCounter` is atomic, so does the `FeedId`, but I'm okay with your idea as we can complete this `per feed basis refactor` first and then think of if we can improve our solution

Reply:
This solution of using `max_debt` creates a new problem —— what is the exact value of `max_debt`? could it be updated and how could we update it?

And it might make this pallet more complex.

But I gonna just pass the `max_debt` as a parameter in `create_feed` momentary.